demo/admin/src/products/ProductForm.tsx

```diff
@@ -71,17 +84,32 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
         id ? { variables: { id } } : { skip: true },
     );
 
-    const initialValues: Partial<FormValues> = data?.product
-        ? {
-              ...filterByFragment<GQLProductFormManualFragment>(productFormFragment, data.product),
-              image: rootBlocks.image.input2State(data.product.image),
-          }
-        : {
-              image: rootBlocks.image.defaultValues(),
-              inStock: false,
-              additionalTypes: [],
-              tags: [],
-          };
+    const initialValues: Partial<FormValues> = React.useMemo<Partial<FormValues>>(() => {
+        const filteredData = data ? filterByFragment<GQLProductFormManualFragment>(productFormFragment, data.product) : undefined;
+        if (!filteredData) {
+            return {
+                image: rootBlocks.image.defaultValues(),
+                inStock: false,
+                additionalTypes: [],
+                tags: [],
+            };
+        }
+        return {
+            ...filteredData,
+            image: rootBlocks.image.input2State(filteredData.image),
+            manufacturerFilter: filteredData.manufacturer
+                ? {
+                      id: filteredData.manufacturer?.addressAsEmbeddable.country,
```

Comment:
don't you need `filteredData.manufacturer?.country,` here?

Reply:
country is nested in the address object, so this is a bit complicated... maybe there is a graphql trick to extract/move a nested field one level up?

Reply:
You could in theory add a field resolver for this. But is it worth it?

Reply:
> But is it worth it?

maybe not. so I'll keep it this way 👍